src/main/java/com/hazelcast/aws/EcsClientStrategy.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.aws;
+
+import com.hazelcast.aws.impl.Constants;
+import com.hazelcast.aws.impl.DescribeNetworkInterfaces;
+import com.hazelcast.aws.impl.DescribeTasks;
+import com.hazelcast.aws.impl.ListTasks;
+import com.hazelcast.aws.utility.Environment;
+import com.hazelcast.aws.utility.MetadataUtils;
+import com.hazelcast.internal.json.Json;
+import com.hazelcast.internal.json.JsonObject;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import static com.hazelcast.aws.impl.Constants.AWS_EXECUTION_ENV_VAR_NAME;
+import static com.hazelcast.aws.impl.Constants.EC2_PREFIX;
+import static com.hazelcast.aws.impl.Constants.HTTPS;
+import static com.hazelcast.aws.utility.StringUtils.isNotEmpty;
+
+/**
+ *
+ */
+class EcsClientStrategy extends AwsClientStrategy {
+
+    private static final String UPPER_ECS = "ECS";
+
+    private static final ILogger LOGGER = Logger.getLogger(AwsClientStrategy.class);
+
+    private String metadataClusterName;
+    private String metadataFamilyName;
+    private String endpointDomain;
+
+    EcsClientStrategy(AwsConfig awsConfig, String endpoint) {
+        super(awsConfig, endpoint);
+        this.endpointDomain = endpoint.substring(Constants.HOSTNAME_PREFIX_LENGTH);
+    }
+
+    @Override
+    public Collection<String> getPrivateIpAddresses() throws Exception {
+        retrieveAndParseMetadata();
```

Comment:
This method has side effects. It changes the state of your class. That is wrong. It would be better if this method returns an object and you later use it, not to modify `metadataClusterName` and `metadataFamilyName`. That would also make this class thread-safe.